### ISSUE TITLE
Shopping cart is restored after payment with another method than MSP

### ISSUE THREAD

**Issue by Tantuss** [MultiSafepay/Magento2Msp#41]
When using other payment method, for example 'Cash on delivery', after a successful order the shopping cart is not empty.
When you add a new product to your shopping cart, it is not added, but the old shopping cart is loaded. Or when you go directly to the shopping cart page, the old products are loaded.

The reason for this, is because in this file:
vendor/multisafepay/magento2msp/etc/frontend/events.xml
There is this:
`  <event name="controller_action_predispatch_checkout_cart_index">
    <observer name="msp_action_predispatch_checkout_cart_index" instance="MultiSafepay\Connect\Model\Observers\RestoreQuote" />
  </event>`

If you then look into the observer: vendor/multisafepay/magento2msp/Model/Observers/RestoreQuote.php

`    public function execute(\Magento\Framework\Event\Observer $observer)
    {
        $session = $this->_objectManager->get('Magento\Checkout\Model\Session');
        $session->restoreQuote();
        return $this;
    }`

So basically, when I return to my shopping cart, there is done a restore of the quote.

Probably this is implemented for MSP, when canceling during payment.

So what should be fixed:
There should be validation if the previous order actually had a payment method related to a payment method of MSP.

**Comment by Jasper-MultiSafepay** [comment]
@Tantuss Thanks for reporting this, we are aware of this issue and will update this/next week which also fixes this issue.

**Comment by cherreman** [comment]
@jasper-multisafepay Do you have a date set for the update? We are currently also experiencing this issue and need to report to our customers.

**Comment by Jasper-MultiSafepay** [comment]
@cherreman I can't communicate an exact date, but the release of the new version will be most likely beginning of next week.

**Comment by cherreman** [comment]
@jasper-multisafepay Hi Jasper, do you have any update on this?

**Comment by Jasper-MultiSafepay** [comment]
@cherreman 
We have had some delays and we're currently testing the plugin.
A quick workaround would be to disable this observer or remove the restorequote call as it's only included for supporting restoring the cart when the back button is pressed from our hosted payment page.
But I will keep you updated when this is fixed.

**Comment by Tantuss** [comment]
Is there a status update?
When can we expect the new version?

**Comment by cherreman** [comment]
@jasper-multisafepay Thanks for the update, but I would rather want to wait until a new release is available. Patching always involves risk and takes time. We consider this a serious issue.

**Comment by Jasper-MultiSafepay** [comment]
@cherreman @Tantuss 
We have solved this and will include this in our next version 1.4.6
According to the developers we will push 1.4.6 tomorrow to Github.

**Comment by Jasper-MultiSafepay** [comment]
@cherreman @Tantuss
We currently have tested and pushed this to the [develop branch](/MultiSafepay/Magento2Msp/tree/develop)
This branch also fixes this issue, so I will be closing this.

**Comment by Tantuss** [comment]
@jasper-multisafepay In my opinion it can only be closed if it is in the master branch. Because by leaving this in the develop branch, you basically say that it is not yet ready for live usage.

When are you going to release it to the master branch?

**Comment by Jasper-MultiSafepay** [comment]
@Tantuss 
In that case I will reopen this and wait for the master branch.
Master branch release will coincide with the plugin release on our website and will be done on Monday/Tuesday.

**Comment by JonkRuud** [comment]
@Tantuss  1.4.6 was just pushed to master and tagged.
